rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala

```diff
@@ -56,6 +53,21 @@ final case class HistoryRepositoryImpl[F[_]: Sync, C, P, A, K](
     Blake2b256Hash.create(joinChannelBits.toByteVector)
   }
 
+  private def hashContinuationsChannels(channels: Seq[C]): Blake2b256Hash = {
+    val chs = channels
+      .map(c => serializeC.encode(c))
+      .sorted(util.ordByteVector)
+    val channelsBits    = codecSeq(codecByteVector).encode(chs).get
+    val joinChannelBits = channelsBits ++ continuationSuffixBits
+    Blake2b256Hash.create(joinChannelBits.toByteVector)
```

Comment:
this logic of creating a continuation channel appears almost identical in at least 3 places